kb(grid):change field delimiter (#1126)
telerik/blazor-docs: knowledge-base/grid-csv-export-change-field-delimiter.md

---

> title: Grid export to CSV - change the default field delimiter (comma)

Reviewer: The word default is not used by the customer, so we can try shortening the title.

```suggestion
title: Grid CSV export - change the comma field delimiter
```

description: How to export the Grid to CSV with a different/custom field delimiter?
type: how-to
page_title: Grid export to CSV - change the default field delimiter (comma)
slug: grid-kb-csv-export-change-field-delimiter
position: 
tags: grid, export, csv, field, delimiter, change, semicolon, comma, custom
ticketid: 1576816
res_type: kb
---

## Environment

> <table>

Reviewer: Use spaces instead of tabs.

Author: Are you referring to the table? This is automatically created section from the admin.

	<tbody>
		<tr>
			<td>Product</td>
			<td>Grid for Blazor</td>
		</tr>
	</tbody>
</table>

## Description
I want to change the default field delimiter when exporting the Grid to CSV file. How to set a different list separator?

I want to use a semicolon field delimiter for the exported CSV Grid instead of comma. How to achieve this?

> >tip At the time of writing (**UI for Blazor 3.5.0**), the Blazor Grid [does not expose a built-in option for configuring the field delimiter](https://feedback.telerik.com/blazor/1577167-option-to-configure-the-field-delimiter-when-exporting-to-csv). You may vote for the enhancement and follow it to get status email updates.

Reviewer: Maybe skip the email part, as it can sound spammy :)

```suggestion
>tip At the time of writing (**UI for Blazor 3.5.0**), the Blazor Grid [does not expose a built-in option for configuring the field delimiter](https://feedback.telerik.com/blazor/1577167-option-to-configure-the-field-delimiter-when-exporting-to-csv). You may vote for the enhancement and follow it to receive status updates.
```

## Solution

> A possible option is to manually modify the exported CSV file before it reaches the client to change the field delimiter. You may handle the [OnAfterExport]({%slug grid-export-events%}#onafterexport) event of the Grid.

Author: ```suggestion
A possible option is to manually modify the exported CSV file before it reaches the client to change the field delimiter. You may handle the [OnAfterExport]({%slug grid-export-events%}#onafterexport) event of the Grid. The `Stream` field of its event argument contains the output of the export as a `MemoryStream`.
```

@using Telerik.Windows.Documents.Spreadsheet.Model
@using System.IO

<TelerikGrid Data="@GridData" Pageable="true">

    <GridToolBar>
        <GridCommandButton Command="CsvExport" Icon="file-csv">Export to CSV</GridCommandButton>
        <label class="k-checkbox-label"><TelerikCheckBox @bind-Value="@ExportAllPages" />Export All Pages</label>
    </GridToolBar>

    <GridExport>
        <GridCsvExport FileName="telerik-grid-export"
                       AllPages="@ExportAllPages"
                       OnAfterExport="@OnCSVAfterExport">
        </GridCsvExport>
    </GridExport>

    <GridColumns>
        <GridColumn Field="@nameof(SampleData.ProductId)" Title="ID" Width="100px" />
        <GridColumn Field="@nameof(SampleData.ProductName)" Title="Product Name" Width="300px" />
        <GridColumn Field="@nameof(SampleData.UnitsInStock)" Title="In stock" Width="100px" />
        <GridColumn Field="@nameof(SampleData.Price)" Title="Unit Price" Width="200px" />
        <GridColumn Field="@nameof(SampleData.Discontinued)" Title="Discontinued" Width="100px" />
        <GridColumn Field="@nameof(SampleData.FirstReleaseDate)" Title="Release Date" Width="300px" />
    </GridColumns>

> </TelerikGrid>

Reviewer: If you want empty lines in the Grid declaration, be consistent :)

```suggestion

</TelerikGrid>
```
